server/internal/usecase/interactor/integration.go

```diff
@@ -24,38 +24,47 @@ func NewIntegration(r *repo.Container) interfaces.Integration {
 	}
 }
 
-func (i Integration) FindByMe(ctx context.Context, operator *usecase.UserOperator) (integration.List, error) {
-	return Run1(ctx, &operator.Operator, i.repos, Usecase().Transaction(),
+func (i Integration) FindByMe(ctx context.Context, operator *usecase.Operator) (integration.List, error) {
+	return Run1(ctx, operator, i.repos, Usecase().Transaction(),
 		func() (integration.List, error) {
-			in, err := i.repos.Integration.FindByUser(ctx, operator.User())
+			if operator.User == nil {
+				return nil, interfaces.ErrInvalidOperator
+			}
+			in, err := i.repos.Integration.FindByUser(ctx, *operator.User)
 			if err != nil {
 				return nil, err
 			}
 			return in, nil
 		})
 }
 
-func (i Integration) FindByIDs(ctx context.Context, ids id.IntegrationIDList, operator *usecase.UserOperator) (integration.List, error) {
-	return Run1(ctx, &operator.Operator, i.repos, Usecase().Transaction(),
+func (i Integration) FindByIDs(ctx context.Context, ids id.IntegrationIDList, operator *usecase.Operator) (integration.List, error) {
+	return Run1(ctx, operator, i.repos, Usecase().Transaction(),
 		func() (integration.List, error) {
+			if operator.User == nil {
+				return nil, interfaces.ErrInvalidOperator
+			}
```

Comment:
You can check before executing a transaction. That way, if there is a problem with the operator, there is no need to start a transaction!

```go
if operator.User == nil {
	return nil, interfaces.ErrInvalidOperator
}

return Run1(ctx, operator, i.repos, Usecase().Transaction(),
		func() (integration.List, error) {
```

Also adding a method for `Usecase()` like `Usecase().UserOnly().Transaction()`,  `Usecase().IntegrationOnly().Transaction()` is good.

```go
return Run1(ctx, operator, i.repos, Usecase().UserOnly().Transaction(),
		func() (integration.List, error) {
```